packages/react-dom/src/events/ReactDOMEventReplaying.js

```diff
@@ -119,6 +119,13 @@ import {IS_REPLAYED, PLUGIN_EVENT_SYSTEM} from './EventSystemFlags';
 import {legacyListenToTopLevelEvent} from './DOMLegacyEventPluginSystem';
 import {listenToTopLevelEvent} from './DOMModernPluginEventSystem';
 import {addResponderEventSystemEvent} from './DeprecatedDOMEventResponderSystem';
+import {
+  DefaultLanePriority,
+  getCurrentUpdateLanePriority,
+  higherLanePriority,
+  setCurrentUpdateLanePriority,
+} from 'react-reconciler/src/ReactFiberLane';
```

Comment:
Shouldn't import stuff from `src/`

Reply:
This is done a few times in this file already, since it's temporary is it fine for now?

Reply:
It looks like we import `react-reconciler/src/ReactFiberReconciler` everywhere and almost never just import from `react-reconciler` except in the NoopRenderer, so I moved this to import from the `ReactFiberReconciler`.

Let me know if that still wrong.

Reply:
Update: switching to `react-reconciler` causes failures due to a dependency cycle:

```
ReactDOMEventReplaying
  => react-reconciler
  => ReactFiberReconciler
  => ReactFiberReconciler.new.js
  => ReactFiberHostConfig
  => ReactDOMHostConfig
  => ReactDOMEventReplaying
```

This causes the host config to only be partially evaluated before it's used in places like [here](https://github.com/facebook/react/blob/a30a1c6ef3a621a57e9372c8249439dbfd9a7375/packages/react-reconciler/src/ReactFiberCompleteWork.new.js#L161), breaking all of the tests.

I think since we're already importing from `react-reconciler/src` here, for now it's safer to not introduce the cycle and import from `react-reconciler/src` again.

Reply:
> I think since we're already importing from react-reconciler/src here, for now it's safer to import from react-reconciler/src again and not introduce the cycle.

This rationale is exactly why I don't want to add more deep imports, because the next person will say "oh, well Rick already did it so it must be ok" 😆

As a concrete example for why deep imports aren't great, imagine if we decide to fork ReactFiberLane.js into separate new and old modules. Then we'd have to also fork the entirety of the React DOM renderer.

The dependency cycle is unfortunate. Longer term, we need to stop performing side effects during module initialization. In ReactFiberCompleteWork, the forking between mutation mode and persistent mode should happen inside of each export implementation, instead of at the module level.

Trying to think of a temporary solution...

Reply:
You could use the same injection strategy as here. Then the workarounds are consistent. https://github.com/facebook/react/blob/f4097c1aef173ea0cb873bc18b47d1ef12bab4b3/packages/react-dom/src/events/ReactDOMEventReplaying.js#L60-L66

But now that I'm looking at that, couldn't all the priority stuff move into `attemptHydrationAtCurrentPriority`? That's a reconciler function, and it's already being injected.

Reply:
Agreed that it's problematic and that I'd be making the problem worse. Would you be up for me filing a task to fix the cycle after this lands? Then I can work on that while we're collecting the warnings that are logged.

If we move it into `attemptHydrationAtCurrentPriority`, how would we get the priority at the time that the hydration was queued?

Reply:
Yeah I guess you'd still need a getter. How about:

- `getCurrentUpdatePriority()`
- `attemptHydrationAtPriority(fn, priority)`

And use injection, like we do for the other reconciler methods in this file

Reply:
If you fix this last thing then I'm comfortable stamping the PR.

The reason I'm being a stickler is because figuring out the layering of where the state (`currentUpdatePriority`) lives is crucial for deciding whether the overall approach is correct / worth it.

Reply:
Sounds good, `attemptHydrationAtPriority` is the same as `ReactFiberReconciler.runWithPriority` anyway.

Reply:
Done, note that `runWithPriority`, `getCurrentUpdatePriority` and `attemptHydrrationAtPriortity` all return/accept the `ReactPriorityLevel` instead of a `LanePriority` with the thinking that `ReactPriorityLevel` is already exposed and we probably don't want to expose Lane implementations until we know for sure that callers will need to know about them.

If you feel strongly that we should expose the LanePriory instead lmk and I can switch them before landing. The internal callsites still all use `setCurrentUpdateLanePriorty` and `getCurrentUpdateLanePriority` which _do_ operate on the LanePriority.